src/components/instructor/AssignmentsView.js

```diff
@@ -9,16 +11,83 @@ import {useLocation} from 'react-router-dom'
 // display a table with columns 
 // assignment id, title, dueDate and buttons to grade, edit, delete each assignment
 
-const AssignmentsView = (props) => {
+const [assignments, setAssignments] = useState([]);
+const [message, setMessage] = useState('');
 
-    const location = useLocation();
-    const {secNo, courseId, secId} = location.state;
-     
-    return(
-        <> 
-           <h3>Not implemented</h3>
-        </>
-    );
-}
+useEffect(() => {
+    fetchAssignments();
+}, []);
 
-export default AssignmentsView;
+const fetchAssignments = async () => {
+    try {
+        const response = await fetch(`${SERVER_URL}/sections/${secNo}/assignments`);
+        if (response.ok) {
+            const assignments = await response.json();
+            setAssignments(assignments);
+        } else {
+            const json = await response.json();
+            setMessage("response error: " + json.message);
+        }
+    } catch (err) {
+        setMessage("network error: " + err);
+    }
+};
+
+const deleteAssignment = async (assignmentId) => {
+    try {
+        const response = await fetch(`${SERVER_URL}/assignments/${assignmentId}`, {
+            method: 'DELETE',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+        });
+        if (response.ok) {
+            setMessage("Assignment deleted successfully");
+            fetchAssignments();
+        } else {
+            const json = await response.json();
+            setMessage("Delete failed: " + json.message);
+        }
+    } catch (err) {
+        setMessage("network error: " + err);
+    }
+};
+
+return (
+    <Paper>
+        <h3>Assignments for Section {secNo}</h3>
+        <h4>{message}</h4>
+        <Table>
+            <TableHead>
+                <TableRow>
+                    <TableCell>Assignment ID</TableCell>
+                    <TableCell>Title</TableCell>
+                    <TableCell>Due Date</TableCell>
+                    <TableCell>Actions</TableCell>
+                </TableRow>
+            </TableHead>
+            <TableBody>
+                {assignments.map((assignment) => (
+                    <TableRow key={assignment.assignmentId}>
+                        <TableCell>{assignment.assignmentId}</TableCell>
+                        <TableCell>{assignment.title}</TableCell>
+                        <TableCell>{assignment.dueDate}</TableCell>
+                        <TableCell>
+                            <Button variant="contained" color="primary" onClick={() => {/* navigate to grade page */}}>
```

Comment:
This button doesn't do anything right now.

Reply:
I'd say leaving it for now should be fine if we plan to implement functionality later.